bundles/CoreBundle/src/DependencyInjection/PimcoreCoreExtension.php

```diff
@@ -413,4 +419,29 @@ private function configureAdapterFactories(ContainerBuilder $container, array $f
 
         $serviceLocator->setArgument(0, $arguments);
     }
+
+    /**
+     * Extract class definitions and prefixes if configuration has more than just a class definition
+     */
+    private function flattenConfigurationForTypeLoader(array $configuration) : array
+    {
+        $newConfiguration = [
+            'map' => [],
+        ];
+        if (array_key_exists('map', $configuration)) {
+            foreach ($configuration['map'] as $type => $config) {
+                if (array_key_exists('class', $config)) {
+                    $newConfiguration['map'][$type] = $config['class'];
+                }
+            }
+        }
+
+        if (array_key_exists('prefixes', $configuration)) {
```

Comment:
I'd rather go for `isset()` here, it's quicker and easier to read.

Reply:
Done.